server/schedule/placement/rule_list.go

```diff
@@ -108,17 +129,27 @@ func buildRuleList(rules map[[2]string]*Rule) (ruleList, error) {
 			sr.deleteRule(p.rule)
 		}
 		if i == len(points)-1 || !bytes.Equal(p.key, points[i+1].key) {
+			var endKey []byte
+			if i != len(points)-1 {
+				endKey = points[i+1].key
+			}
+
```

Comment:
Why move these code?

Reply:
Just for reusing. It is not a big deal to write it twice, though. Do I need to change?

Reply:
I think it's ok to update like this.